packages/shared/src/hooks/useOrganizationList.tsx

```diff
@@ -1,39 +1,248 @@
+        };
+
+  const canFetch = !!(clerk.loaded && user);
+
+  // Some gymnastics to adhere to the rules of hooks
+  // We need to make sure useSWR is called on every render
+  const fetchInvitations = !clerk.loaded
+    ? () => ({ data: [], total_count: 0 } as ClerkPaginatedResponse<UserOrganizationInvitationResource>)
+    : () => user?.getOrganizationInvitations(paginatedParams);
+
+  const {
+    data: userInvitationsData,
+    isValidating: userInvitationsValidating,
+    isLoading: userInvitationsLoading,
+    error: userInvitationsError,
+    mutate: userInvitationsMutate,
+  } = useSWR(
+    !triggerInfinite && canFetch && paginatedParams ? cacheKey('userInvitations', user, paginatedParams) : null,
+    fetchInvitations,
+    { keepPreviousData: internalKeepPreviousData },
+  );
+
+  const getInfiniteKey = (
+    pageIndex: number,
+    previousPageData: ClerkPaginatedResponse<UserOrganizationInvitationResource> | null,
+  ) => {
+    if (!canFetch || !paginatedParams || !triggerInfinite) {
+      return null;
+    }
+
+    return cacheKey('userInvitations', user, {
+      initialPage: initialPageRef.current + pageIndex,
+      initialPageSize: initialPageSizeRef.current,
+      status: internalStatus,
+    });
+  };
+
+  const {
+    data: userInvitationsDataInfinite,
+    isLoading: userInvitationsLoadingInfinite,
+    isValidating: userInvitationsInfiniteValidating,
+    error: userInvitationsInfiniteError,
+    size,
+    setSize,
+    mutate: userInvitationsInfiniteMutate,
+  } = useSWRInfinite(getInfiniteKey, ({ initialPage, initialPageSize, status }) => {
+    return !clerk.loaded || !user
+      ? ({ data: [], total_count: 0 } as ClerkPaginatedResponse<UserOrganizationInvitationResource>)
+      : user.getOrganizationInvitations({
+          initialPage,
+          initialPageSize,
+          status,
+        });
+  });
+
+  const isomorphicPage = useMemo(() => {
+    if (triggerInfinite) {
+      return size;
+    }
+    return paginatedPage;
+  }, [triggerInfinite, size, paginatedPage]);
+
+  const isomorphicSetPage: CustomSetAction<number> = useCallback(
+    numberOrgFn => {
+      if (triggerInfinite) {
+        void setSize(numberOrgFn);
+        return;
+      }
+      return setPaginatedPage(numberOrgFn);
+    },
+    [setSize],
+  );
+
+  const isomorphicData = useMemo(() => {
+    if (triggerInfinite) {
+      return userInvitationsDataInfinite?.map(a => a?.data).flat() ?? [];
+    }
+    return userInvitationsData?.data ?? [];
+  }, [triggerInfinite, userInvitationsDataInfinite, userInvitationsData]);
+
+  const isomorphicCount = useMemo(() => {
+    if (triggerInfinite) {
+      return userInvitationsDataInfinite?.[userInvitationsDataInfinite?.length - 1]?.total_count || 0;
+    }
+    return userInvitationsData?.total_count ?? 0;
+  }, [triggerInfinite, userInvitationsDataInfinite, userInvitationsData]);
+
+  const isomorphicIsLoading = triggerInfinite ? userInvitationsLoadingInfinite : userInvitationsLoading;
```

Comment:
❓ What does isomorphic mean in this context? It feels that it's a bit confusing.

Reply:
I needed a way to introduce parity between the returned values of useSWR and useSWRInfinite.

based on the params received in the hook return the appropriate values

Reply:
The naming still feels a bit off. I'd just remove the isomorphic prefix.

Reply:
Sure, i'll update this shortly in a following PR